src/pathfinding.h

```diff
@@ -0,0 +1,45 @@
+#ifndef PATHFINDING_H
+#define PATHFINDING_H
+
+enum pf_special : int {
```

Comment:
If you're really concerned about cache coherency, you can knock this down to an enum : char.